### PR TITLE
Added python version and example

### PR DESCRIPTION
**Pull request by sye8**
The filter has been implemented in Python
For both Python and Java version, a main method is included as an example

### CODE REVIEW

File: contrib/java/KalmanFilter.java [hunk 1]
```diff
@@ -87,4 +85,18 @@ public void setMeasurementNoise(double noise){
 	public void setProcessNoise(double noise){
 		    this.R = noise;
 	}
+    
+    /**
+      * Main method for usage example
+      *
+      */
+    public static void main(String[] args){
```
Comment (wouterbulten) [inline contrib/java/KalmanFilter.java]: Please remove this method. Examples can be placed in a README.md but should not be present in the core class.

File: contrib/python/kalman.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    B = 0
+    C = 1
+
+    R = 0
+    Q = 0
+
+    cov = float('nan')
+    x = float('nan')
+
+    # Constructor
```
Comment (wouterbulten) [inline contrib/python/kalman.py]: Please use valid docstrings at all functions. You can for example use the numpy standard (http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).

File: contrib/python/kalman.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# license GNU LESSER GENERAL PUBLIC LICENSE v3
+# author Sifan Ye
+#
+# see https://github.com/wouterbulten/kalmanjs
+#
+
+import math
+
+class KalmanFilter:
+    A = 1
```
Comment (wouterbulten) [inline contrib/python/kalman.py]: These variables should not be declared static but should be class members. Please move them to the constructor.

File: contrib/python/kalman.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    B = 0
+    C = 1
+
+    R = 0
+    Q = 0
+
+    cov = float('nan')
```
Comment (wouterbulten) [inline contrib/python/kalman.py]: Maybe it is better to use None here. But, please let me know if you think nan would be more appropriate.

File: contrib/python/kalman.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    def set_measurement_noise(self, noise):
+        self.Q = noise
+
+    # Sets process noise
+    #
+    # noise: The new measurement noise
+    def set_process_noise(self, noise):
+        self.R = noise
+
+# Main method as an example
```
Comment (wouterbulten) [inline contrib/python/kalman.py]: This should be moved to a README.md file (if you want to include an example)

**Review by wouterbulten — CHANGES_REQUESTED** [review]
Thank you for your pull request. It would be a nice addition to have a python implementation of the filter in the contrib folder.

I do have some comments on your implementation though. Please see my comments and try to change your implementation to address these.

Let me know if you have any questions.

File: contrib/python/kalman.py (hunk 5 above)
Comment (sye8) [reply contrib/python/kalman.py]: Has been moved

**Comment by wouterbulten** [comment]
Hi @ysf199711, would you accept licensing the python version under the MIT? I want to publish the newest version of the JS library using the MIT license to make it a bit more usable for closed-source projects.

**Comment by sye8** [comment]
Yes, I can accept that

**Review by wouterbulten — APPROVED** [review]

**Comment by wouterbulten** [comment]
Great! I will merge the branch later this week (together with the license change).

**Pull request merged**